include/aspect/particle/generator/random_function.h

```diff
@@ -0,0 +1,120 @@
+
+#include <deal.II/base/parsed_function.h>
+
+#include <boost/random.hpp>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Generator
+    {
+      /**
+       * Generate random distribution of particles over entire simulation domain.
+       * The density of particles is determined by a user-defined probability
+       * density function.
+       */
+      template <int dim>
+      class RandomFunction : public Interface<dim>, public SimulatorAccess<dim>
+      {
+        public:
+          /**
+           * Constructor.
+           *
+           */
+          RandomFunction();
+
+          /**
+           * Generate a uniformly randomly distributed set of particles in the current triangulation.
+           */
+          virtual
+          void
+          generate_particles(World<dim> &world);
+
+          /**
+           * Declare the parameters this class takes through input files.
+           */
+          static
+          void
+          declare_parameters (ParameterHandler &prm);
+
+          /**
+           * Read the parameters this class declares from the parameter file.
+           */
+          virtual
+          void
+          parse_parameters (ParameterHandler &prm);
+
+        private:
+          /**
+           * Random number generator and an object that describes a
+           * uniform distribution on the interval [0,1]. These
+           * will be used to generate particle locations at random.
+           */
+          boost::mt19937            random_number_generator;
+          boost::uniform_01<double> uniform_distribution_01;
+
+          /**
+           * Number of particles to create
+           */
+          particle_index n_tracers;
+
+          /**
+           * A function object representing the temperature.
+           */
+          Functions::ParsedFunction<dim> function;
+
+          /**
+           * Generate a set of particles uniformly randomly distributed within the
+           * specified triangulation. This is done using "roulette wheel" style
+           * selection weighted by cell volume. We do cell-by-cell assignment of
+           * particles because the decomposition of the mesh may result in a highly
+           * non-rectangular local mesh which makes uniform particle distribution difficult.
+           *
+           * @param [in] cells Map between accumulated cell weight and cell index
+           * @param [in] global_weight The integrated probability function
+           * @param [in] start_weight The starting weight of the first cell of the local process.
+           * @param [in] num_particles The total number of particles to generate.
+           * @param [in] start_id The starting ID to assign to generated particles of the local process.
+           * @param [inout] world The particle world the particles will exist in
+           *
+           */
+          void uniform_random_particles_in_subdomain (const std::map<double,LevelInd> &cells,
```

Comment:
(Or is the confusion simply due to copy-pasting this function and its documentation from the `UniformRandom` class?)